api/app.go

```diff
@@ -306,17 +313,53 @@ func (app *App) AddError() {
 	app.Errors.Update(1)
 }
 
+func buildAddress(port int) string {
+	if port == 0 {
+		return "localhost:"
+	} else {
+		return fmt.Sprintf("localhost:%d", port)
+	}
+}
+
 // Start starts listening for web requests at specified host and port
 func (app *App) Start() error {
 	l := app.Logger.With(
 		zap.String("source", "app"),
 		zap.String("operation", "Start"),
 		zap.String("host", app.Host),
-		zap.Int("port", app.Port),
+		zap.Int("port", app.HTTPPort),
 	)
 
+	grpcLis, err := net.Listen("tcp", buildAddress(app.GRPCPort))
+	if err != nil {
+		return fmt.Errorf("error trying to listen for connections: %v", err)
+	}
+	app.GRPCEndpoint = grpcLis.Addr().String()
+
+	httpLis, err := net.Listen("tcp", buildAddress(app.HTTPPort))
+	if err != nil {
+		return fmt.Errorf("error listening on HTTPEndpoint: %v", err)
+	}
+	app.HTTPEndpoint = httpLis.Addr().String()
+
+	//errch is the channel for retrieving errors from server goroutines.
+	errch := make(chan error, 2)
+	var wg sync.WaitGroup
+	wg.Add(2)
+
+	go func() {
+		app.startGRPCServer(grpcLis, errch)
+		wg.Done()
+	}()
+	go func() {
+		app.startHTTPServer(httpLis, errch)
+		wg.Done()
+	}()
+
+	stopped := make(chan bool, 1)
 	go func() {
-		app.App.Run(app.Engine)
+		wg.Wait()
+		stopped <- true
```

Comment:
At line 367 below:

I don't think we should capture `syscall.SIGKILL` - if it gets to that, then the app should just die, no meddling on our part. The others are fine.

TBH I don't even think that works.

Reply:
Added TODO to verify that this works.

Reply:
Please remove SIGKILL from the list. I'm fairly certain we shouldn't try to capture it.